pages/_app.tsx

```diff
@@ -30,8 +32,15 @@ const HeaderWrapper = styled.div`
 
 function MyApp({ Component, pageProps }: AppProps): JSX.Element {
     const pageTitle = pageProps.pageTitle || 'Documentation';
+    const navData = pageProps.navDot ? pageProps.navDot[0] : [];
     const [showSidebar, setShowSidebar] = useState(true);
     const [showSideToc, setShowSideToc] = useState(false);
+    const [breadCrumb, setBreadcrumb] = useState([]);
+    const router = useRouter();
+    const docPage = router.asPath.split('/')[2];
+    useEffect(() => {
+        setBreadcrumb(searchBreadCrump(navData.dotcmsdocumentationchildren, docPage));
+    }, []);
```

Comment:
Why `useEffect`?

Reply:
To make sure the `setBreadCrumb` just runs once, the first time the page is open. we only care that the `sidenav` appears open on the first page the user visits.